src/accelerate/utils/transformer_engine.py

```diff
@@ -20,16 +21,23 @@
 from .operations import GatheredParameters
 
 
-if is_fp8_available():
-    import transformer_engine.pytorch as te
+def require_te(fn):
+    if not is_fp8_available():
+        @wraps(fn)
+        def i(*args, **kwargs):
+            raise ImportError(f"Using `{fn.__name__}` requires transformer_engine to be installed.")
+        return i
+
+    return fn
```

Comment:
Let's not overcomplicate things. Keep the `if/else` in the original functions, + import during their calls. We follow this method since those are the user-facing APIs that eventually get called.